sound/soc/intel/boards/bdw_rt286.c

```diff
@@ -183,16 +183,30 @@ static struct snd_soc_dai_link card_dai_links[] = {
 	},
 };
 
+#define CODEC_DAI_NAME "rt286-aif1"
+
 static int card_suspend_pre(struct snd_soc_card *card)
 {
-	struct snd_soc_dai *codec_dai = snd_soc_card_get_codec_dai(card, "rt286-aif1");
+	struct snd_soc_dai *codec_dai = snd_soc_card_get_codec_dai(card, CODEC_DAI_NAME);
+
+	if (!codec_dai) {
+		dev_err(card->dev, "Codec DAI %s not found for card %s", CODEC_DAI_NAME,
+			card->name);
+		return -EINVAL;
```

Comment:
but should we return an error? Can we flip the check and say if (codec_dai) return snd_soc_component_set_jack() else return 0 instead?

Reply:
It's true that there's no point in preventing the suspend from happening.

I would also not add a dev_err() for something that cannot really happen.

````c
if (codec_dai) 
    return snd_soc_component_set_jack(codec_dai->component, NULL, NULL);
return 0;
````

Reply:
> You should edit the commit message to mention "all existing uses of snd_soc_card_get_codec_dai() in Intel machine drivers do check for a NULL pointer, bdw_rt286 is the exception to the rule".

@plbossart No machine driver checks a NULL pointer if `snd_soc_card_get_codec_dai()` is called in `card_suspend_pre/card_resume_post`.  All other uses of `snd_soc_card_get_codec_dai()` check for a NULL pointer, and return -EIO.  This is also true for avs/boards.

@crojewsk-intel  mentioned that "It's impossible to get NULL at this point in time."

@plbossart A clean up you mentioned above is good but looks not very necessary. I think we can close the PR and mark non-issue for static analysis tool?

Reply:
@aiChaoSONG all other drivers use a different method to get the component, with a loop on the codec name. In the end it's the same thing.

I am fine with doing nothing and marking the non-issue in the tool.